src/main/scala/is/hail/methods/LinearRegression.scala

```diff
@@ -19,7 +19,11 @@ object LinearRegression {
     require(vds.wasSplit)
 
     val (y, cov, completeSamples) = RegressionUtils.getPhenoCovCompleteSamples(vds, yExpr, covExpr)
-    val sampleMask = vds.sampleIds.map(completeSamples.toSet).toArray
+    val completeSamplesSet = completeSamples.toSet
```

Comment:
All in general three variables (completeSampleSet, sampleMask and cmopleteSampleIndex) are needed below.  I tried it, but it felt like more trouble than it's worth.  I made the other suggested changes.